### PR TITLE
Convert motion gateway/handler to use protobuf/isotp

### PR DESCRIPTION
**Pull request by bonnetn**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -146,10 +131,16 @@
         i.provide('lidar_adapter', RPLIDARAdapter)
 
     if simulation or stub_socket_can:
-        i.provide('socket_can_adapter', StubSocketAdapter)
+        i.provide('socket_can_adapter', LoopbackSocketAdapter)
+        i.provide('motor_board_adapter', LoopbackSocketAdapter)
     else:
-        reader, writer = await _tcp_reader_writer('localhost', 1200)
-        i.provide('socket_can_adapter',
+        # reader, writer = await _tcp_reader_writer('localhost', 1200)
```
Comment (tictacmenthe) [inline src/main.py:137]: why keep the comments 👎

File: src/robot/adapter/socket/socket_adapter.py [hunk 2]
```diff
@@ -36,33 +38,45 @@ def __init__(self, reader: StreamReader, writer: StreamWriter):
...
+                               raw_payload=raw_payload,
+                               packet_number_recv=self.packet_count_recv)
 
             for handler in self.handlers:
                 await handler(raw_payload)
 
     async def send(self, data: bytes) -> None:
-        self.writer.write(data)
+        hex_payload = binascii.hexlify(data)
+        self.packet_count_sent += 1
```
Comment (tictacmenthe) [inline src/robot/adapter/socket/socket_adapter.py:51]: maybe only count the packet as sent after the end of transmission? (in case it fails)
Do you manage socket write errors?

File: src/robot/handler/protobuf.py [hunk 3]
```diff
@@ -24,12 +27,16 @@ async def dispatch_message(self, bus_message: BusMessage) -> None:
         type_msg = bus_message.WhichOneof("message_content")
         if type_msg == "heartbeat":
             pass
```
Comment (tictacmenthe) [inline src/robot/handler/protobuf.py:29]: suggestion : if a heartbeat isn't received for a board for a few seconds, the board has an issue, log it.

File: src/robot/handler/protobuf.py [hunk 4]
```diff
@@ -24,12 +27,16 @@ def __init__(self, match_action_controller: MatchActionController):
...
+        elif type_msg == "movementEnded":
+            self.localization_controller.movement_done()
```
Comment (tictacmenthe) [inline src/robot/handler/protobuf.py:31]: also pass the "blocked" parameter of the movement_done message? even if it is not used yet

File: src/util/can_id.py [hunk 5]
```diff
@@ -1,6 +0,0 @@
-"""
```
Comment (tictacmenthe) [inline src/util/can_id.py:1]: This file was so useful ! 💯

File: src/robot/adapter/socket/socket_adapter.py (hunk 2 above)
Comment (Melanie-Co) [reply src/robot/adapter/socket/socket_adapter.py:51]: +1

Comment (Melanie-Co) [reply src/robot/adapter/socket/socket_adapter.py:51]: After the drain

File: src/robot/handler/protobuf.py [hunk 6]
```diff
@@ -24,12 +27,16 @@ async def dispatch_message(self, bus_message: BusMessage) -> None:
...
         elif type_msg == "encoderPosition":
-            if bus_message.encoderPosition.right_tick and bus_message.encoderPosition.left_tick:
-                pass
+            self.localization_controller.update_odometry_position(
+                bus_message.encoderPosition.left_tick,
+                bus_message.encoderPosition.right_tick)
```
Comment (Melanie-Co) [inline src/robot/handler/protobuf.py:35]: What do you do if you do not have all the parameters ?

File: src/robot/handler/protobuf_test.py [hunk 7]
```diff
@@ -2,12 +2,43 @@
...
+    """
+    bus_message = BusMessage(encoderPosition=EncoderPositionMsg(
+        left_tick=1,
+        right_tick=-2,
+    ))
+    msg_bytes = bus_message.SerializeToString()
+    await protobuf_handler.translate_message(msg_bytes)
+
+    localization_controller_mock.update_odometry_position.assert_called_once_with(
+        1, -2)
```
Comment (Melanie-Co) [inline src/robot/handler/protobuf_test.py:36]: + test with left-tick or right_tick or both missing ?

File: src/simulation/gateway/simulation.py [hunk 8]
```diff
@@ -16,31 +14,30 @@ class SimulationGateway:
...
-            can_id.PROPULSION_ENCODER_POSITION,
-            packet.encode_propulsion_encoder_position(
-                packet.PropulsionEncoderPositionPacket(
-                    left_tick=left_tick,
-                    right_tick=right_tick,
-                )))
+        bus_message = BusMessage(encoderPosition=EncoderPositionMsg(
+            left_tick=left_tick,
+            right_tick=right_tick,
+        ))
```
Comment (Melanie-Co) [inline src/simulation/gateway/simulation.py:38]: You should keep the last value if a parameter is None or seems invalid.

File: src/robot/handler/protobuf.py (hunk 3 above)
Comment (bonnetn) [reply src/robot/handler/protobuf.py:29]: Should be in a separate commit, created an issue https://github.com/outech-robotic/hl-flowing-clean-arch/issues/88

File: src/robot/handler/protobuf.py (hunk 4 above)
Comment (bonnetn) [reply src/robot/handler/protobuf.py:31]: Yes, done

File: src/main.py (hunk 1 above)
Comment (bonnetn) [reply src/main.py]: :'( Fine it's removed

File: src/util/can_id.py (hunk 5 above)
Comment (bonnetn) [reply src/util/can_id.py:1]: It was

File: src/robot/handler/protobuf.py (hunk 6 above)
Comment (bonnetn) [reply src/robot/handler/protobuf.py:36]: I always have all the parameters.
It's the encoder responsability to put all the necessary fields in the message. If a field is not provided, protobuf libraries usually assume that the caller want to pass the zero value in the field.

TLDR: no risk of exception here

File: src/robot/handler/protobuf_test.py (hunk 7 above)
Comment (bonnetn) [reply src/robot/handler/protobuf_test.py:38]: It's not possible

File: src/simulation/gateway/simulation.py (hunk 8 above)
Comment (bonnetn) [reply src/simulation/gateway/simulation.py:38]: Parameter should never be None, unless we use the "well-known" types. But since we are using regular ints, it will always be populated

File: src/robot/adapter/socket/socket_adapter.py (hunk 2 above)
Comment (bonnetn) [reply src/robot/adapter/socket/socket_adapter.py]: Will add counters in another commit
